Userland/Applications/Presenter/Presentation.cpp

```diff
@@ -183,16 +189,23 @@ ErrorOr<DeprecatedString> Presentation::render()
     }
 </style><script>
     function goto(slideIndex, frameIndex) {
-        // FIXME: Honor the frameIndex.
-        let slide;
-        for (slide of document.getElementsByClassName("slide")) {
-            slide.style.display = "none";
+        for (let slide of document.getElementsByClassName("slide")) {
+          slide.style.display = "none";
+        }
+        for (let frame of document.getElementsByClassName("frame")) {
+          frame.style.display = "none";
+        }
+
+        let slide = document.getElementById(`slide${slideIndex}`);
+        if (slide) slide.style.display = "block";
+
+        for (let i = 0; i <= frameIndex; i++) {
+          let frame = document.getElementById(`slide${slideIndex}-frame${i}`);
+          if (frame) frame.style.display = "block";
```

Comment:
i thought that you might want to work on classes here, i.e. switching not the styles on the specific elements but adding/removing classes instead (e.g. `visible`, `active` or something along those lines). this could provide a greater control of visuals enabling other styling if needed etc.